src/awkward/_v2/contents/regulararray.py

```diff
@@ -694,6 +681,14 @@ def mergeable(self, other, mergebool):
         ):
             return self._content.mergeable(other.content, mergebool)
 
+        # For n-dimensional NumpyArrays, let's now convert them to RegularArray
+        # We could add a special case that tries to first convert self to NumpyArray
+        # and merge conventionally, but it's not worth it at this stage.
+        elif (
+            isinstance(other, ak._v2.contents.numpyarray.NumpyArray)
+            and other.purelist_depth > 1
+        ):
+            return self._content.mergeable(other.toRegularArray().content, mergebool)
```

Comment:
This `toRegularArray()` is going to happen twice for every merger. It might be possible to do this at an earlier stage of processing and therefore only do it once.

But not for this PR.

Reply:
Yes, this is not ideal at all. I suspect we'd replace `mergeable` interface with `prepare_merge(self, other)`, and if the two are mergeable, return potentially modified layouts.

I think that's a bigger change than we want to make in this PR, which is already "big".